app/views/admin/show.html.erb

```diff
@@ -0,0 +1,25 @@
+<h1><%=@app.name%>'s Application (Admin View)</h1>
+<p><u>Address</u>: <%=@app.address%>, <%=@app.city%>, <%=@app.zip%></p>
+<p><u>Description</u>: <%=@app.description%></p>
+<p><u>Status</u>: <%=@app.status%></p>
+
+<p><h2><u>Pets on this application</u></h2></p> 
+<% @app.pets.each do |pet| %>
+  <%= link_to "#{pet.name}", "/pets/#{pet.id}" %> 
+  
+  <%if @app.status == "Accepted" %>
+    <p>your application for <%=pet.name %> has been Approved!</p>
+
+  <%elsif @app.status == "Rejected"%>
+    <p>your application for <%=pet.name %> has been Rejected!</p>
+
+  <% else @app.status == "Pending" %>
+    <%= button_to "approve application to adopt #{pet.name}", "/admin/apps/#{@app.id}?approved=true", method: :patch%></p>
```

Comment:
A more RESTful route here would include pets in the path to indicate that you are updating the joins record between applications and pets. Consider something like "/applications/:application_id/pets/:pet_id"

Reply:
and then you can still have a query parameter for "approved" or "rejected"!